src/main/java/com/bmw/hmm_lib/ViterbiAlgorithm.java

```diff
@@ -157,7 +159,9 @@ private boolean hmmBreak(Map<S, Double> message) {
     private ForwardStepResult forwardStep(HmmProbabilities<S, O> hmmProbabilities,
             TimeStep<S, O> prevTimeStep, TimeStep<S, O> curTimeStep, Map<S, Double> message) {
         final ForwardStepResult result = new ForwardStepResult(curTimeStep.candidates.size());
-        assert( !prevTimeStep.candidates.isEmpty());
```

Comment:
I use assertions to document preconditions of nonpublic methods, invariants or postconditions as recommended [here](http://docs.oracle.com/javase/8/docs/technotes/guides/language/assert.html). Moreover, I enable assertions when running unit or regression tests, so they are actually checked and hence serve a similar purpose as Junit assertions.